From 4fdad4f38b121966ef4627cc966a9cac10e4cc97 Mon Sep 17 00:00:00 2001
Message-ID: <4fdad4f38b121966ef4627cc966a9cac10e4cc97.1745272461.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1745272461.git.sharlatanus@gmail.com>
References: <cover.1745272461.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Mon, 21 Apr 2025 18:37:33 +0100
Subject: [PATCH 24/32] gnu: python-stsci-stimage: Update to 0.3.0.

* gnu/packages/astronomy.scm (python-stsci-stimage): Update to 0.3.0.
[arguments] <build-backend>: Use default.
<test-flags>: Remove them.
<phases>: Remove 'waf-configure-build; add 'build-extensions.
[native-inputs]: Remove python-waf.

Change-Id: Iaefe173dc00084ca5a751e01bb2e88e1ef5ad6fe
---
 gnu/packages/astronomy.scm | 26 +++++++++-----------------
 1 file changed, 9 insertions(+), 17 deletions(-)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index f6fef80900..cbab2c3daf 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -7096,33 +7096,25 @@ (define-public python-stsci-skypac
 (define-public python-stsci-stimage
   (package
     (name "python-stsci-stimage")
-    (version "0.2.10")
-    (source (origin
-              (method url-fetch)
-              (uri (pypi-uri "stsci_stimage" version))
-              (sha256
-               (base32
-                "02qzblw11kfdw9rp2m26zbzlzl3w0dnrhncn383a3sw3dwjn9lpf"))))
+    (version "0.3.0")
+    (source
+     (origin
+       (method url-fetch)
+       (uri (pypi-uri "stsci_stimage" version))
+       (sha256
+        (base32 "10vhc2kfryis37k5jkg357z7lhlmyci4makzy50xgh08648ak7cd"))))
     (build-system pyproject-build-system)
     (arguments
      (list
-      #:build-backend "setuptools.build_meta"
-      #:test-flags #~(list "test_c")
       #:phases
       #~(modify-phases %standard-phases
-          ;; Test steps are taken from GitHub Actions, see
-          ;; <https://github.com/spacetelescope/stsci.stimage/issues/27>
-          (add-before 'check 'waf-configure-build
+          (add-before 'check 'build-extensions
             (lambda _
-              (copy-file (string-append
-                          #$(this-package-native-input "python-waf") "/bin/waf")
-                         "waf")
-              (invoke "python" "waf" "configure" "build"))))))
+              (invoke "python" "setup.py" "build_ext" "--inplace"))))))
     (native-inputs
      (list python-pytest
            python-setuptools
            python-setuptools-scm
-           python-waf
            python-wheel))
     (propagated-inputs
      (list python-numpy))
-- 
2.49.0

